### PR TITLE
Return json for API key errors

### PR DESCRIPTION
**Pull request by kalkov**
#### Summary
Return authentication errors in JSON: 
```json
{"errors": {"detail": "Error message"}} 
```

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/graphql/context_plug.ex [hunk 1]
```diff
@@ -37,19 +37,25 @@ defmodule SanbaseWeb.Graphql.ContextPlug do
     |> put_private(:absinthe, %{context: context})
   end
 
+  defp build_error_msg(msg) do
+    %{errors: %{detail: msg}} |> Jason.encode!()
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/context_plug.ex]:
```suggestion
    %{errors: %{details: msg}} |> Jason.encode!()
```

File: lib/sanbase_web/graphql/context_plug.ex [hunk 2]
```diff
@@ -37,19 +37,25 @@ defmodule SanbaseWeb.Graphql.ContextPlug do
...
-        Logger.warn(warn_msg)
+        msg = "Bad authorization header: #{error}"
+        Logger.warn(msg)
+
+        response_msg = build_error_msg(msg)
 
         conn =
           conn
-          |> send_resp(400, warn_msg)
+          |> send_resp(400, response_msg)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/context_plug.ex]: you can just put `|> send_resp(400, build_error_msg(msg))` ehere

File: lib/sanbase_web/graphql/context_plug.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/context_plug.ex]: Also you need to change the tests for this. All other places use plural for details

Comment (kalkov) [reply lib/sanbase_web/graphql/context_plug.ex]: It's "detail" for 500: https://github.com/santiment/sanbase2/blob/master/lib/sanbase_web/views/error_view.ex

Comment (kalkov) [reply lib/sanbase_web/graphql/context_plug.ex]:
```json
{"errors":{"detail":"Internal server error"}}
```

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/context_plug.ex]: I don't think the code you linked is used anywhere

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/context_plug.ex]: The "details" we see in errors are of the format described here: https://github.com/absinthe-graphql/absinthe/blob/master/guides/errors.md#basic-errors

Comment (tspenov) [reply lib/sanbase_web/graphql/context_plug.ex]: BTW isn't it better to return the same structure as absinthe but without locations:

```
{
  "errors": [ {"message": "message 1"}, ...]
}
```
@kalkov @IvanIvanoff

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/context_plug.ex]: @kalkov You were right - when the Plug parser breaks (invalid json) then the ErrorView handles the response, so this `detail` you linked is the one you see

**Pull request merged**